### PR TITLE
ON-12323 # Updated validation and docs

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: src/lib/forms-schema/property-schemas.ts [hunk 1]
```diff
@@ -204,3 +204,13 @@ export const conditionallyShowSchemas = {
 export const storageType = Joi.string()
   .label('Storage type')
   .valid(['legacy', 'public', 'private'])
+
+const regexPattern = Joi.string()
+const regexFlags = Joi.string().regex(/^[dgimsuy]*$/)
```
Comment (mymattcarroll) [inline src/lib/forms-schema/property-schemas.ts:209]:
```suggestion
const regexFlags = Joi.string().regex(/^[dgimsuy]+$/)
```

File: src/lib/forms-schema/property-schemas.ts [hunk 2]
```diff
@@ -204,3 +204,13 @@ export const conditionallyShowSchemas = {
...
+const regexMessage = Joi.string()
```
Comment (mymattcarroll) [inline src/lib/forms-schema/property-schemas.ts:210]: This should be required if `regexPattern` has a value.

**Comment by mymattcarroll** [comment]
Can I be a massive jerk and ask for a jest test?

**Comment by Zaxist** [comment]
> Can I be a massive jerk and ask for a jest test?

Man, what a massive jerk.

File: tests/forms-schema.test.ts [hunk 3]
```diff
@@ -5514,3 +5514,103 @@ describe('Date and Time `NOW` option', () => {
     )
   })
 })
+
+describe.only('Regex Custom Validation Properties', () => {
```
Comment (mymattcarroll) [inline tests/forms-schema.test.ts:5518]:
```suggestion
describe('Regex Custom Validation Properties', () => {
```

**Pull request merged**